src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs

```diff
@@ -208,6 +208,75 @@ public static void UnsafeAwaitAwaiterFromRuntimeAsync<TAwaiter>(TAwaiter awaiter
             SuspendAsync2(sentinelContinuation);
             return;
         }
+
+        // TODO: should this be called "AwaitFromRuntimeAsync" ?  (i.e. same as above, but no "Awaiter")
+        //
+        // Marked intrinsic since this needs to be
+        // recognizes as an async2 call.
+        [Intrinsic]
+        [BypassReadyToRun]
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static unsafe T Await<T>(Task<T> task)
+        {
+            // TODO: handle complete tasks more efficiently.
+            //if (!task.IsCompleted)
+            //{
+                ref RuntimeAsyncAwaitState state = ref t_runtimeAsyncAwaitState;
+                Continuation? sentinelContinuation = state.SentinelContinuation;
+                if (sentinelContinuation == null)
+                    state.SentinelContinuation = sentinelContinuation = new Continuation();
+
+                Continuation myContinuation = new Continuation();
+                myContinuation.GCData = new object[] { task };
+                myContinuation.Resume = &AwaitHelper<T>.Resume;
+
+                state.Notifier = task.GetAwaiter();
+                sentinelContinuation.Next = myContinuation;
+
+                //     RETURN {default(T), myContinuation}
+                //
+                SuspendAsync2(myContinuation);
+
+                // unreachable
+                return task.ResultOnSuccess;
+            //}
+            //else
+            //{
+            //    //   RETURN {task.Result, null}
+            //    //
+            //    T result = task.Result;
+            //    ReturnAsync2(Unsafe.AsPointer(ref result));
```

Comment:
Just guessing how `ReturnAsync2` could look. 

If generic intrisic is ok, it could also be just `ReturnAsync2<T>(result)`

Reply:
I don't think we need an intrinsic, this can just be a normal return.

Reply:
Would a regular return guarantee that the continuation return is null?

Reply:
It would need the VM to tell the JIT this function has runtime-async calling convention. That's probably something we'd want anyway. It might also be the only thing we need to allow using `UnsafeAwaitAwaiterFromRuntimeAsync` in its implementation.

Reply:
I will try. We do not do that for `UnsafeAwaitAwaiterFromRuntimeAsync`, but that may work just because it is a simpler method.